src/ensembl/src/content/app/entity-viewer/gene-view/components/transcripts-filter/TranscriptsFilter.tsx

```diff
@@ -14,52 +14,118 @@
  * limitations under the License.
  */
 
-import React, { useState } from 'react';
+import React, { useEffect } from 'react';
 import classNames from 'classnames';
-import noop from 'lodash/noop';
 import { connect } from 'react-redux';
-import { RootState } from 'src/store';
+
 import { isEntityViewerSidebarOpen } from 'src/content/app/entity-viewer/state/sidebar/entityViewerSidebarSelectors';
+import {
+  getFilters,
+  getSortingRule
+} from 'src/content/app/entity-viewer/state/gene-view/transcripts/geneViewTranscriptsSelectors';
+import {
+  setFilters,
+  Filters,
+  setSortingRule,
+  SortingRule
+} from 'src/content/app/entity-viewer/state/gene-view/transcripts/geneViewTranscriptsSlice';
 
-import { ReactComponent as CloseIcon } from 'static/img/shared/close.svg';
-import { ReactComponent as ChevronUp } from 'static/img/shared/chevron-up.svg';
 import RadioGroup, {
   RadioOptions
 } from 'src/shared/components/radio-group/RadioGroup';
 
-import styles from './TranscriptsFilter.scss';
 import Checkbox from 'src/shared/components/checkbox/Checkbox';
 
+import { ReactComponent as CloseIcon } from 'static/img/shared/close.svg';
+import { ReactComponent as ChevronUp } from 'static/img/shared/chevron-up.svg';
+
+import { Transcript } from 'src/content/app/entity-viewer/types/transcript';
+import { RootState } from 'src/store';
+
+import styles from './TranscriptsFilter.scss';
+
 type Props = {
+  filters: Filters;
+  sortingRule: SortingRule;
   toggleFilter: () => void;
   isSidebarOpen: boolean;
+  transcripts: Transcript[];
+  setFilters: (filters: Filters) => void;
+  setSortingRule: (sortingRule: SortingRule) => void;
 };
 
 type OptionValue = string | number | boolean;
 
-const radioData: RadioOptions = [
-  { value: 'default', label: 'Default' },
-  { value: 'length_longest', label: 'Spliced length: longest - shortest' },
-  { value: 'length_shortest', label: 'Spliced length: shortest - longest' }
+const sortingOrderToLabel = [
+  [SortingRule.DEFAULT, 'Default'],
+  [
+    SortingRule.SPLICED_LENGTH_LONGEST_TO_SHORTEST,
+    'Spliced length: longest – shortest'
+  ],
+  [
+    SortingRule.SPLICED_LENGTH_SHORTEST_TO_LONGEST,
+    'Spliced length: shortest – longest'
+  ]
 ];
 
+const radioData: RadioOptions = sortingOrderToLabel.map(([key, value]) => ({
+  value: key,
+  label: value
+}));
+
 const TranscriptsFilter = (props: Props) => {
-  const [isChecked, setChecked] = useState(false);
+  const biotypes = props.transcripts
+    .map((a) => a.biotype)
+    .filter(Boolean) as string[];
+  const uniqueBiotypes = Array.from(new Set(biotypes));
+  // TODO: Add protein coding options in RadioOptions if there are protein coding biotype
+  // Initial filters as object,  pass to redux store (dispatch action passing initial filters)
```

Comment:
The second line of the comment is no longer necessary. As for the first line of the comment, remind me, are we doing anything about it?

Reply:
We still need to do the protein coding for MAP but needed to confirm with Andrea the behavior.